src/System.Data.SqlClient/src/System/Data/SqlClient/TdsParserStaticMethods.cs

```diff
@@ -61,12 +61,14 @@ internal static int GetCurrentProcessIdForTdsLoginOnly()
         {
             if (s_currentProcessId == NoProcessId)
             {
-                // In ProjectK\CoreCLR we don't want to take a dependency on an assembly
-                // just to grab the real Process Id that the server doesn't really use
-                // So, instead, pick a random number and use that for all connections
-                Random rand = new Random();
-                int processId = rand.Next();
-                Threading.Interlocked.CompareExchange(ref s_currentProcessId, processId, NoProcessId);
+                // Pick up the process Id from the current process instead of randomly generating it.
+                // This would be helpful while tracing application related issues.
+                int processId;
+                using (System.Diagnostics.Process p = System.Diagnostics.Process.GetCurrentProcess())
+                {
+                    processId = p.Id;
+                }
+                System.Threading.Volatile.Write(ref s_currentProcessId, processId);
```

Comment:
Nit: looking at it again now, there isn't even a need for the local `processId`... the write could just be moved into the using block.  But this is fine.